File: frontend/src/types/activity.ts
export type ActivityCategory =
  | 'SPORTS'
  | 'MOVIES'
  | 'DINING'
  | 'OUTDOORS'
  | 'GAMING'
  | 'STUDY'
  | 'MUSIC'
  | 'OTHER';

export type ActivityStatus = 'OPEN' | 'FULL' | 'CANCELLED' | 'COMPLETED';

export interface CreatorInfo {
  id: number;
  displayName: string;
}

export interface Activity {
  id: number;
  title: string;
  location: string;
  activityDate: string;
  category: ActivityCategory;
  status: ActivityStatus;
  maxParticipants: number;
  currentParticipants: number;
  creator: CreatorInfo;
  createdAt: string;
}

export interface ParticipantInfo {
  userId: number;
  displayName: string;
  joinedAt: string;
}

export interface ActivityDetail extends Activity {
  description: string | null;
  updatedAt: string;
  participants: ParticipantInfo[];
}

File: backend/src/main/java/com/whosup/entity/Activity.java
package com.whosup.entity;

import jakarta.persistence.*;
import java.time.Instant;

@Entity
@Table(name = "activities")
public class Activity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 200)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false, length = 300)
    private String location;

    @Column(name = "activity_date", nullable = false)
    private Instant activityDate;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private ActivityCategory category = ActivityCategory.OTHER;

    @Column(name = "max_participants", nullable = false)
    private int maxParticipants;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private ActivityStatus status = ActivityStatus.OPEN;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "creator_id", nullable = false)
    private User creator;

    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = Instant.now();
        updatedAt = Instant.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = Instant.now();
    }

    public Activity() {}

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }

    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }

    public String getLocation() { return location; }
    public void setLocation(String location) { this.location = location; }

    public Instant getActivityDate() { return activityDate; }
    public void setActivityDate(Instant activityDate) { this.activityDate = activityDate; }

    public ActivityCategory getCategory() { return category; }
    public void setCategory(ActivityCategory category) { this.category = category; }

    public int getMaxParticipants() { return maxParticipants; }
    public void setMaxParticipants(int maxParticipants) { this.maxParticipants = maxParticipants; }

    public ActivityStatus getStatus() { return status; }
    public void setStatus(ActivityStatus status) { this.status = status; }

    public User getCreator() { return creator; }
    public void setCreator(User creator) { this.creator = creator; }

    public Instant getCreatedAt() { return createdAt; }

    public Instant getUpdatedAt() { return updatedAt; }
}
